filebeat/prospector/log/harvester.go

```diff
@@ -326,7 +344,7 @@ func (h *Harvester) SendStateUpdate() {
 
 	d := util.NewData()
 	d.SetState(h.state)
-	h.forwarder.Outlet.OnEvent(d)
+	h.publishState(d)
```

Comment:
The events ACKs are directly forwarded to the registrar. But as the harvester and the stateOutlet share the same beat.Client instance, the events will be forwarded in correct order to the registrar.